lib/ansible/galaxy/api.py

```diff
@@ -853,10 +853,18 @@ def get_collection_versions(self, namespace, name):
             try:
                 modified_date = self.get_collection_metadata(namespace, name).modified_str
             except GalaxyError as err:
-                if err.http_code != 404:
+                if err.http_code == 404:
+                    # No collection found, return an empty list to keep things consistent with the various APIs
+                    return []
+                elif err.http_code != 500 or 'v3' in self.available_api_versions:
                     raise
-                # No collection found, return an empty list to keep things consistent with the various APIs
-                return []
+
+                # v2 appears to have a bug on some versions, where the metadata URL is an HTTP error 500
+                # since this URL is only used to handle the cache, proceed as if the cache needs to be refreshed.
+                modified_date = None
+                if '%s.%s' % (namespace, name) in modified_cache:
+                    del modified_cache['%s.%s' % (namespace, name)]
+                    self._set_cache()
```

Comment:
@webknjaz  Since v2 GalaxyAPI is deprecated and close to removal, I'm second-guessing this. Can we just remove the v2 tests early, instead of adding code to support the lack of fully functional v2 in 4.9.2?

Or, could we keep using an older galaxy ng for v2 coverage  until 2.20 development starts?

Reply:
@s-hertel I think that coverage is a good reason to keep this. But also, I was looking forward to backporting this to reduce the CI flakiness. I feel like this might be okay to backport, but removals wouldn't be.

Should we merge+backport this and then delete whatever in follow-ups when 2.20 comes? The main value in this for me is the CI stability, which spans across several branches IIRC.

cc @mattclay would this be acceptable to backport for the sake of CI?

Reply:
Since this is a bug fix for V2, it seems like a good idea to backport it. We should continue testing V2 for versions that support it. Once we branch for 2.20 we'll be able to rip out all the V2 code and tests.

Reply:
If v2 was used by anything other than old galaxy and custom servers mimicking the v2 endpoint, I'd be more in favor of backporting this. This seems featureish for 3rd party galaxy servers, which should conform with https://old-galaxy.ansible.com v2 or GalaxyNG v3.

IMO we should not be adding more code for v2, unless it fixes issues with  https://old-galaxy.ansible.com.

Reply:
Dunno, I haven't looked deeper. I was originally hoping for just an image update, but these changes grew on top, as you and sivel suggested.